apps/web/components/pages/task/IssueCard.tsx

```diff
@@ -1,43 +1,62 @@
 /* eslint-disable no-mixed-spaces-and-tabs */
-import { Card, Modal, SpinnerLoader, Text } from 'lib/components';
+import {
+	Card,
+	Dropdown,
+	DropdownItem,
+	Modal,
+	SpinnerLoader,
+	Text,
+} from 'lib/components';
 import { TaskInput, TaskLinkedIssue } from 'lib/features';
 import { useRecoilValue } from 'recoil';
 import { detailedTaskState } from '@app/stores';
-import { IHookModal, useModal, useQuery, useTeamTasks } from '@app/hooks';
+import {
+	IHookModal,
+	useModal,
+	useQuery,
+	useSyncRef,
+	useTeamTasks,
+} from '@app/hooks';
 import { ITeamTask, TaskRelatedIssuesRelationEnum } from '@app/interfaces';
 import { useTranslation } from 'lib/i18n';
 import { useCallback, useMemo, useState } from 'react';
 import { createTaskLinkedIsssueAPI } from '@app/services/client/api';
 import { clsxm } from '@app/utils';
 import { ChevronDownIcon, ChevronUpIcon, PlusIcon } from 'lib/components/svgs';
 
-const IssueCard = ({ related }: { related: boolean }) => {
+export const RelatedIssueCard = () => {
+	const { trans } = useTranslation();
 	const modal = useModal();
+
 	const task = useRecoilValue(detailedTaskState);
 	const { tasks } = useTeamTasks();
 	const [hidden, setHidden] = useState(false);
 
+	const { actionType, actionTypeItems, onChange } = useActionType();
+
 	const linkedTasks = useMemo(() => {
-		return (
+		const issues =
 			task?.linkedIssues
```

Comment:
@paradoxe35 can you try to improve this logic? multiple .filter and .map will reduce the performance when we have more tasks.